### ISSUE TITLE
Default list selection breaks Spotify on iOS and Android devices

### ISSUE THREAD

**Issue by vaguerant** [pi-hole/pi-hole#2084]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [ x ] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [ x ] The issue I am reporting can be *replicated*.
- [ x ] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**
1

---
**Expected behaviour:**
The Spotify app should work on smart devices.

**Actual behaviour:**
The app throws errors when doing basically anything, e.g. searching, as access to the server is being blocked.

**Steps to reproduce:**
Install the Spotify app on an iOS/Android device, register or log in to a Spotify account, try to do things.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**
N/A.

**Troubleshooting undertaken, and/or other relevant information:**
What's going on here is that StevenBlack/hosts is integrating FadeMind/hosts.extras, which in turn is blocking `spclient.wg.spotify.com`, a domain that is used both to serve advertisements to desktop and mobile users and to provide most of the actual functionality in the mobile app. The desktop/web version of Spotify does not seem to need this domain to function correctly.

I understand that "whether or not to break Spotify in the hosts list" is an upstream issue. I am not filing an issue to suggest that Pi-Hole solve this upstream issue. There is an ongoing discussion of this issue at StevenBlack/hosts#552, although it is marked wontfix further upstream at FadeMind/hosts.extras#13.

The issue I am filing here specifically relates to whether it is desirable for Pi-Hole to include by default a hosts list which breaks a major service like Spotify for a substantial number of its users--basically any Spotify users in a Pi-Hole house(-or-other-establishment-)hold with a mobile or tablet device.

Obviously there is the quasi-solution of "instruct users to whitelist servers to un-break services included in default lists (and also explain to them that whitelisting is broken in current stable if that's still true at time of reading)". Is that the most desirable solution for Pi-Hole? Should Pi-Hole's **default functionality** be to **break web services** and then instruct users **who are tech-savvy enough to recognize what is even going wrong** on how to fix this issue? Is it OK that some segment of users **won't figure this out at all and will either deal with broken services or abandon Pi-Hole because it broke their desirable web services**?

Sorry for all the bold text, just trying to draw attention to the big questions that this touches upon for a project like Pi-Hole. I'm not strictly advocating for a particular solution, although I do think Pi-Hole's maintainers should consider deeply which lists do or do not align with the goals of Pi-Hole and whether breaking services by default and requiring end-users to fix them is within the bounds of Pi-Hole's goals.

Depending on what happens upstream, this issue may or may not resolve itself to the satisfaction of Pi-Hole's maintainers. However, as it currently stands, Pi-Hole is "choosing" to break Spotify by default by including a default hosts list which breaks Spotify. Is this OK **right now**? If upstream decides this is OK in an ongoing fashion, is that OK for Pi-Hole in an ongoing fashion?

**Comment by DL6ER** [comment]
> If upstream decides this is OK in an ongoing fashion, is that OK for Pi-Hole in an ongoing fashion?

I guess so. There are good reasons for why we do not deal with blacklist creation ourselves although it may seem obvious that we could create our own one. The current mode of operation is that users see issues and open issue reports with the original blocking list creators. I agree that not everyone will be able to narrow down which list leads to breaking which service (although it is not too complicated) but I think it is fair to assume that someone who is willing to contact at least us will find out, eventually.

> The issue I am filing here specifically relates to whether it is desirable for Pi-Hole to include by default a hosts list which breaks a major service like Spotify for a substantial number of its users--basically any Spotify users in a Pi-Hole house(-or-other-establishment-)hold with a mobile or tablet device.

It is (in general) a great list and leads to a very broad coverage of various ad providers. It is out of our control if any of the upstream lists decides to block things also like Windows Update and alike. There are discussions going on whether or not we should complicate the install process by giving users the option to chose which blocking lists to include in their installation, but I don't think that this will lead to any substantial improvement of the situation and will just complicate the installation a bit more. Yes, users will (maybe) see, understand and remember that what we are blocking comes from external blocking lists but then we obviously have to also provide a "recommended" set of lists and I'm very much in favor of adding StevenBlack/hosts into this list.

To summarize: No, I don't think we should or could change something fundamentally as it is out of our control what the various upstream providers do and we decided long ago that we don't want to e.g. provide a whitelist, regularly generated by us, to fix upstream issues. As you may have seen, we even abandoned the long standing feature to auto-whitelist the domains of the blocking lists so that they couldn't block each other to avoid infantilizing our users.
